common/frame.go

```diff
@@ -1498,40 +1498,59 @@ func (f *Frame) Press(selector string, key string, opts goja.Value) {
 	applySlowMo(f.ctx)
 }
 
+// SelectOption selects the given options and returns the array of
+// option values of the first element found that matches the selector.
 func (f *Frame) SelectOption(selector string, values goja.Value, opts goja.Value) []string {
 	f.log.Debugf("Frame:SelectOption", "fid:%s furl:%q sel:%q", f.ID(), f.URL(), selector)
 
-	parsedOpts := NewFrameSelectOptionOptions(f.defaultTimeout())
-	if err := parsedOpts.Parse(f.ctx, opts); err != nil {
-		k6ext.Panic(f.ctx, "%w", err)
+	popts := NewFrameSelectOptionOptions(f.defaultTimeout())
+	if err := popts.Parse(f.ctx, opts); err != nil {
+		k6ext.Panic(f.ctx, "parse: %w", err)
+	}
+	v, err := f.selectOption(selector, values, popts)
+	if err != nil {
+		k6ext.Panic(f.ctx, "selectOption: %w", err)
 	}
 
-	fn := func(apiCtx context.Context, handle *ElementHandle) (interface{}, error) {
+	applySlowMo(f.ctx)
+
+	return v
+}
+
+func (f *Frame) selectOption(selector string, values goja.Value, opts *FrameSelectOptionOptions) ([]string, error) {
+	selectOption := func(apiCtx context.Context, handle *ElementHandle) (interface{}, error) {
 		return handle.selectOption(apiCtx, values)
 	}
-	actFn := f.newAction(
-		selector, DOMElementStateAttached, parsedOpts.Strict, fn, []string{},
-		parsedOpts.Force, parsedOpts.NoWaitAfter, parsedOpts.Timeout,
+	act := f.newAction(
+		selector, DOMElementStateAttached, opts.Strict, selectOption,
+		[]string{}, opts.Force, opts.NoWaitAfter, opts.Timeout,
 	)
-	value, err := callApiWithTimeout(f.ctx, actFn, parsedOpts.Timeout)
+	v, err := callApiWithTimeout(f.ctx, act, opts.Timeout)
 	if err != nil {
-		k6ext.Panic(f.ctx, "%w", err)
+		return nil, err
 	}
-
-	arrayHandle, ok := value.(api.JSHandle)
+	selectHandle, ok := v.(jsHandle)
 	if !ok {
-		k6ext.Panic(f.ctx, "%w", err)
+		return nil, fmt.Errorf("unexpected select element type %T", v)
 	}
-	properties := arrayHandle.GetProperties()
-	strArr := make([]string, 0, len(properties))
-	for _, property := range properties {
-		strArr = append(strArr, property.JSONValue().String())
-		property.Dispose()
+
+	// gets <option> values inside <select>.
+	optHandles, err := selectHandle.getProperties()
+	if err != nil {
+		return nil, fmt.Errorf("getProperties: %w", err)
+	}
+	vals := make([]string, 0, len(optHandles))
+	for _, oh := range optHandles {
+		vals = append(vals, oh.JSONValue().String())
+		if err := oh.dispose(); err != nil {
+			return nil, fmt.Errorf("optionHandle.dispose: %w", err)
+		}
+	}
```

Comment:
Thanks for the review, @imiric. It filters out the ones in the `ElementHandle` method here:

https://github.com/grafana/xk6-browser/blob/8c774d89122e19ddc71fe6bb12ee91cc767acfa9/common/element_handle.go#L472-L568

Yeah, let's change the comment here.

And yeah, I'm also confused about the naming 🤦 That's why I added a comment there as it might be confusing to others as it confused me every time I see it. Are you sure we shouldn't change it? The only thing that worries me is that maybe it's a general concept, IDK.

Reply:
Hhmm no, that converts the input `values` to an array `injected_script.js` can work with. The actual filtering is done [here](https://github.com/grafana/xk6-browser/blob/8c774d89122e19ddc71fe6bb12ee91cc767acfa9/common/js/injected_script.js#L628-L652). Anyway, it's clear now, thanks.

OK, let's update this comment and the `SelectOption` description as well.

I wouldn't change our internal method name, as that would introduce more confusion from being different to CDP's. So let's leave it as is, but maybe add a comment in our wrapper that clarifies it.

Reply:
Yup, it's where the action happens 👍  Please see: dc1c9a6